iceoryx_binding_c/source/c_publisher.cpp

```diff
@@ -99,11 +99,31 @@ void iox_pub_deinit(iox_pub_t const self)
     self->~cpp2c_Publisher();
 }
 
-iox_AllocationResult iox_pub_loan_chunk(iox_pub_t const self, void** const chunk, const uint32_t payloadSize)
+bool iox_pub_configure_custom_header(iox_pub_t const self,
+                                     const uint32_t customHeaderSize,
+                                     const uint32_t customHeaderAlignment)
 {
-    auto result = PublisherPortUser(self->m_portData)
-                      .tryAllocateChunk(payloadSize, CHUNK_DEFAULT_PAYLOAD_ALIGNMENT)
-                      .and_then([&chunk](ChunkHeader* h) { *chunk = h->payload(); });
+    auto settingsCheck = iox::mepoo::ChunkSettings::create(0U, 1U, customHeaderSize, customHeaderAlignment);
+    if (settingsCheck.has_error())
+    {
+        return false;
+    }
+
```

Comment:
Logic seems a bit weird/redundant since in the true case settingsCheck is not used anymore. The way I understand it this is just to check validity of the arguments before setting them in the publisher.

Reply:
yes, the actual check happens when the user calls `iox_pub_loan_chunk` but since configuring the custom header with this function might lead to wrong parameters, with this check it is ensured that with `iox_pub_loan_chunk` only the payload parameter can cause an error